### ISSUE TITLE
Updating the Xtext Community Website

### ISSUE THREAD

**Issue by miklossy** [borisbrodski/Jnario#10]
Hello Jnario team,

The Xtext team would like to update the [Xtext community website](https://www.eclipse.org/Xtext/community.html) listing all the cool projects people are building around Xtext/Xtend.

See also the corresponding [GitHub issue](https://github.com/eclipse/xtext/issues/1541) and [Bugzilla ticket](https://bugs.eclipse.org/bugs/show_bug.cgi?id=328477).

If you are interested that the Jnario project is listed there, please provide the following information as a comment to this issue:

```
<tr>
	<td><a href="project_link">project_name</a></td>
	<td>project_description_max_2_sentences</td>
	<td>license</td>
	<td>category</td>
	<td>author(s)</td>
</tr>

```
We will then update the Xtext community website and get it online as soon as possible.

Thanks!

See also [issue](https://github.com/sebastianbenz/Jnario/issues/174) in the original Jnario repository.

**Comment by riederm** [comment]
Hi Tamas, thx for your interest!

Maybe we can use this issue for a discussion. My situation is as follows:

In the past, I supported Boris in making Jnario a pure Xbase-based language and I think we did a good job - we even backported Xtend-Richstrings to Xbase and we have now a Jnario which only (or mostly) uses API-classes from Xtext. Therefore it can be migrated to new Xtext versions pretty easily (we did this several times now in my company with only minor issues).

On the other hand, I tried multiple times to "fix" the _Scenarios_ and the _Suites_. The Scoping in both of them needs to be rewritten - it uses proxy-resolving a lot and therefore build-times expload in big workspaces - to a point that in my company we stopped writing scenarios and suites. 

Also, the semantics of _Scenarios_ is broken in my opinion. When re-using _Steps_ from other _Scenarios_ the code generation does weird things that quickly result in wrong behavior (e.g. all local variables and the before-block is copied to your local Scenario - it's very hard to keep track of what variables you actually see and what you don't - also duplicate local fields are a problem, etc.).

So for me, only the _Facts_ are usable at the moment - in my opinion, the scoping and validation would also require some love since it's performance also degenerates pretty quickly. But with only the specs left, JUnit 5 and Java11 and above offers more and more of the Jnario features.

to sum it up, I think Jnario is not really compelling at the moment, it needs:
- rewrite scoping and validation for basically all 3 languages
- repair Scenarios and introduce clear and concise behavior when using steps across several Scenarios (even if we would break compatibility!)
- re-write to generate Junit 5 code and get rid of the custom test-runners

are there other opinions?

**Comment by miklossy** [comment]
Hi Mathias!

Thanks for the detailed description about the current state of the Jnario project!

**Comment by borisbrodski** [comment]
Thank you very much for this issue and the discussion so far!

I'm starting to port Jnario to the lastest Xtend now. We are using Specs and Suites, so I'm not as skilled in the Feature part.

@riederm Could you please explain a little bit more, why we should rewrite scoping for all 3 languages?

**Comment by miklossy** [comment]
Hi @borisbrodski !

Thanks for the information, I am really happy hearing that you are porting Jnario to the latest Xtend now. Let me know if there is already a branch about this work.

**Comment by riederm** [comment]
> @riederm Could you please explain a little bit more, why we should rewrite scoping for all 3 languages?

Hey @borisbrodski , maybe I was not very precise with my elaborations.
I think there is some room for improvements, maybe not in scoping itself (there is almost no jnario-scoping code) but in avoiding resolving proxies (in validation, in syntax-highlighting, etc.)

My collegues tend to avoid jnario because of poor perceived performance once we get big workspaces.

I did some quick profiling and it looks like lots of stuff is also tied to richStrings - idk. for now the teams in my company avoid it because of performance.
![image](https://user-images.githubusercontent.com/3087112/75169991-b9c20f80-5729-11ea-92af-3f6b5509aeed.png)

**Comment by borisbrodski** [comment]
Ported to Xtext 2.20.0.

Next I would like to add a Gradle plugin.

@riederm Thank you for the explanations. We have exactly the same situation with the colleagues avoiding Jnario because of the poor performance. I have already implemented some tweaks improving performance for "=> [ ... ]" closures. This hack reduced the build speed and responsiveness by ~25% (see a667f6a2d2f171189fd0597bdece27412ffcbce6)
Are you using this improvement in your Jnario installation?

**Comment by miklossy** [comment]
Nice work @borisbrodski !

But back to my original question: If you agree that the Jnario project is listed on the Xtext community website, please provide the necessary information requested above. I really find that the Jnario project deserves to be listed there, but I need your agreement to do that.

PS: I like the recorded EclipseCon sessions about your Jnario talk, I think we should also add a link to that in the description section.

**Comment by borisbrodski** [comment]
@miklossy Thank you!

Just chatted with the Jnario author @sebastianbenz. It would be great to be listen on the Xtext community website.

```
<tr>
	<td><a href="http://jnario.org/">Jnario</a></td>
	<td>Executable Specifications for Java.</td>
	<td>Eclipse Public License v1.0</td>
	<td>Testing, Test-driven Development (TDD)</td>
	<td>Sebastian Benz, Birgit Engelmann</td>
</tr>
```

I think, after we will release the new version of Jnario, that will work out of the box with the current and many previous versions of Xtext and Eclipse, we could add some more authors on the list. ;)

My videos and talks:

- http://xtextcasts.org/episodes/9-jnario-test-framework
  (Same on YouTube: https://youtu.be/j-IouNAkGvw)
- Eclipse CON 2014
  - YouTube: https://www.youtube.com/watch?v=jCgPfxaY8XQ
  - EclipseCON: https://www.eclipsecon.org/europe2014/session/write-cool-scalable-enterprise-application-tests-xtend-embedded-dsls.html

**Comment by miklossy** [comment]
Nice!

**Comment by miklossy** [comment]
The `Jnario` project has been added to the [Xtext community website](https://www.eclipse.org/Xtext/community.html).
